src/redux/global/configSlice.ts

```diff
@@ -30,27 +36,46 @@ const configSlice = createSlice({
                 timestamp: new Date().toISOString(),
             });
         },
+        setTheme: (state, action: PayloadAction<ThemeConfig>) => {
+            state.theme = action.payload;
+        },
     },
     extraReducers: (builder) => {
         builder.addCase(initConfig.fulfilled, (state, action) => action.payload);
     },
 });
 
-export const { pushRecentProject } = configSlice.actions;
-export const configReducer = configSlice.reducer;
-
 const LOCAL_STORAGE_KEY = 'ace_recent_projects';
 
+export const { pushRecentProject, setTheme } = configSlice.actions;
+export const configReducer = configSlice.reducer;
+
 export const localStorageMiddleware: Middleware = (store) => (next) => (action) => {
     next(action);
 
-    const { recentProjects } = store.getState().config;
+    const { recentProjects, theme } = store.getState().config;
     if (recentProjects !== undefined) {
         localStorage.setItem(
             LOCAL_STORAGE_KEY,
             JSON.stringify(recentProjects),
         );
     }
+
+    if (theme !== undefined) {
+        localStorage.setItem(
+            THEME_LOCAL_STORAGE_KEY,
+            JSON.stringify(theme),
+        );
+    }
+};
+
+export const logger: Middleware = (store) => (next) => (action) => {
+    console.group(action.type);
+    console.info('dispatching', action);
```

Comment:
Can you format it a bit nicer? Capitalize the sentences, use template strings, colons all that jazz

Reply:
Honestly just used it for convenience during debugging, so I'll remove it. A logger should prob be added in a separate pr.

Reply:
As you see fit. I thought it can be useful for bug reports